### PR TITLE
Add ReduceLROnPlateau lr scheduler

### PR DESCRIPTION
**Pull request by kapiblue**

### CODE REVIEW

**Comment by kapiblue** [comment]
ReduceLROnPlateau reduces the learning rate based on validation loss (per epoch).

A nice [blog post about schedulers](https://medium.com/data-scientists-diary/guide-to-pytorch-learning-rate-scheduling-b5d2a42f56d4).

We can also consider other types of schedulers or even Batch-Level Scheduling (since our batch size is large).
